contracts/PositionManagerWETH.sol

```diff
@@ -56,12 +44,13 @@ contract PositionManagerWETH is IPositionManagerWETH, PositionManagerDependent {
         }
 
         if (isCollateralIncrease && collateralChange > 0) {
-            wETH.deposit{ value: msg.value }();
+            IWETH(address(_underlyingCollateralToken)).deposit{ value: msg.value }();
+            wrappedCollateralToken.depositFor(address(this), msg.value);
             collateralChange = msg.value;
```

Comment:
I know it was not introduced in this PR, but maybe require that `collateralChange == msg.value` instead of `collateralChange = msg.value`?

Reply:
Yeah, make sense, I'll do that in the new PR.